incubator/logstash/templates/deployment.yaml

```diff
@@ -38,6 +38,8 @@ spec:
               value: {{ .Values.elasticsearch.port | quote }}
             - name: SYSLOG_PORT
               value: {{ .Values.service.internalPort | quote }}
+            - name: LS_JAVA_OPTS
+              value: "-Xms{{ .Values.initialHeapSize }} -Xmx{{ .Values.maxHeapSize }}"
```

Comment:
It's problematic to set these to such high values by default. I would leave the defaults empty and only set them when specified. Default resources are not set either which is good, i. e. the specified vaules are much higher than is available by default.

Reply:
Good point.
I set them that high because that is the default in the container image.
I will set it up to only set them if they are not empty.

Reply:
Shouldn't be that hard to do. You could create a helper template for that.